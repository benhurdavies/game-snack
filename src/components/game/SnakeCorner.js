import React from 'react';
import { Rect } from 'react-konva';

const SnakeCorner = ({ x, y, blockSize, theme }) => (
  <Rect
    x={x * blockSize}
    y={y * blockSize}
    width={blockSize}
    height={blockSize}
    fill="green"
  />
);

export default SnakeCorner;
