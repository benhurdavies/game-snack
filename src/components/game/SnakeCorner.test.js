import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import SnakeCorner from './SnakeCorner';

describe('Component: SnakeCorner', () => {
  let props = {
    x: 10,
    y: 10,
    blockSize: 20,
    theme: { snakeColor: 'black' },
    corner: 'left-down',
  };

  it('basic shape check', () => {
    const wrapper = shallow(<SnakeCorner {...props} />);
    expect(wrapper.find('Shape').length).toEqual(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<SnakeCorner {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
