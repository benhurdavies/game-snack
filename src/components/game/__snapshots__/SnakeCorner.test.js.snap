// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SnakeCorner renders correctly 1`] = `
<Shape
  sceneFunc={[Function]}
  x={200}
  y={200}
/>
`;
